cargo-shuttle/src/lib.rs

```diff
@@ -486,203 +684,173 @@ impl Shuttle {
-
-                            trace!("installing shuttle-runtime");
-                            std::process::Command::new("cargo")
-                                .arg("install")
-                                .arg("shuttle-runtime")
-                                .arg("--bin")
-                                .arg("shuttle-next")
-                                .arg("--features")
-                                .arg("next")
-                                .output()
-                                .expect("failed to install the shuttle runtime");
-                        };
-                    };
-
-                    runtime_path
-                } else {
-                    trace!(path = ?executable_path, "using alpha runtime");
-                    executable_path.clone()
+                if signal_received {
+                    break;
                 }
-            };
-
-            let (mut runtime, mut runtime_client) = runtime::start(
-                is_wasm,
-                runtime::StorageManagerType::WorkingDir(working_directory.to_path_buf()),
-                &format!("http://localhost:{provisioner_port}"),
-                None,
-                run_args.port - (1 + i) as u16,
-                runtime_path,
-            )
-            .await
-            .map_err(|err| {
-                provisioner_server.abort();
+            } else {
+                runtimes.push(
+                    self.spin_local_runtime(
+                        &run_args,
+                        service,
+                        &provisioner_server,
+                        i as u16,
+                        provisioner_port,
+                    )
+                    .await?,
+                );
+            }
+        }
 
-                err
-            })?;
-
-            let service_name = service.service_name()?;
-
-            let load_request = tonic::Request::new(LoadRequest {
-                path: executable_path
-                    .into_os_string()
-                    .into_string()
-                    .expect("to convert path to string"),
-                service_name: service_name.to_string(),
-                resources: Default::default(),
-                secrets,
-            });
-            trace!("loading service");
-            let response = runtime_client
-                .load(load_request)
-                .or_else(|err| async {
-                    provisioner_server.abort();
-                    runtime.kill().await?;
-
-                    Err(err)
-                })
-                .await?
-                .into_inner();
-
-            if !response.success {
-                error!(error = response.message, "failed to load your service");
-                exit(1);
+        // If prior signal received is set to true we must stop all the existing runtimes and
+        // exit the `local_run`.
+        if signal_received {
+            for mut rt in runtimes {
```

Comment:
Perhaps de-structuring here would be cleaner, something like this?
```suggestion
            for mut (runtime, client) in runtimes {
```

Reply:
Done.